File: src/components/lecture_card.py
import flet as ft
from models.lecture import LectureStatus

class LectureCard(ft.Container):
    def __init__(self, lecture, update_callback):
        super().__init__()
        self.lecture = lecture
        self.update_callback = update_callback 
        
        self.border_radius = 12
        self.padding = 10
        self.margin = ft.margin.only(bottom=10)
        self.bgcolor = self.lecture.course_color
        
        self.shadow = ft.BoxShadow(
            spread_radius=1, blur_radius=5, color="black12", offset=ft.Offset(0, 2)
        )

        title = ft.Text(self.lecture.title, weight="w600", size=15, color="black87", no_wrap=True, overflow=ft.TextOverflow.ELLIPSIS)
        
        time_room = ft.Row([
            ft.Image(src="icons/access_time.svg", width=14, height=14, color="black54"),
            ft.Text(f"{self.lecture.start_time} - {self.lecture.end_time}", color="black54", size=12),
            ft.Container(width=10),
            ft.Image(src="icons/location_on.svg", width=14, height=14, color="black54"),
            ft.Text(self.lecture.room, color="black54", size=12)
        ], spacing=4, alignment=ft.MainAxisAlignment.START)

        self.content = ft.Column([
            title, time_room, ft.Divider(height=1, color="black12"), self.build_status_icons()
        ], spacing=6)

    def build_status_icons(self):
        def create_icon(icon_name, status_value, tooltip_text, active_color):
            is_active = (self.lecture.status == status_value)
            
            # התיקון: שימוש ב-Container לחיץ במקום IconButton
            return ft.Container(
                content=ft.Image(
                    src=f"icons/{icon_name}.svg", 
                    width=20, height=20, 
                    color=active_color if is_active else "black38"
                ),
                tooltip=tooltip_text,
                bgcolor="white" if is_active else "transparent",
                border_radius=20,
                padding=8,
                on_click=lambda e: self.set_status(status_value)
            )

        return ft.Row([
            create_icon("check_circle", LectureStatus.ATTENDED, "הלכתי", "green"),
            create_icon("smart_display", LectureStatus.WATCHED_RECORDING, "הקלטה", "blue"),
            create_icon("pending", LectureStatus.NEEDS_WATCHING, "להשלים", "orange"),
            create_icon("cancel", LectureStatus.SKIPPED, "דילגתי", "red"),
            create_icon("block", LectureStatus.CANCELLED, "בוטלה", "grey700"),
        ], alignment=ft.MainAxisAlignment.SPACE_BETWEEN)

    def set_status(self, new_status):
        self.lecture.status = new_status
        if self.update_callback:
            self.update_callback()

File: flutter_app/lib/features/dashboard/lecture_card.dart
import 'package:flutter/material.dart';
import '../../core/ui/app_icons.dart';
import '../../l10n/app_localizations.dart';
import '../../models/schedule_models.dart';
import 'dashboard_utils.dart';

class LectureCard extends StatelessWidget {
  const LectureCard({
    super.key,
    required this.lecture,
    required this.allLectures,
    required this.showMeetingNumber,
    required this.l10n,
    required this.onStatus,
  });

  final Lecture lecture;
  final List<Lecture> allLectures;
  final bool showMeetingNumber;
  final AppLocalizations l10n;
  final void Function(Lecture, LectureStatus) onStatus;

  static const _compactBtnStyle = ButtonStyle(
    visualDensity: VisualDensity.compact,
    tapTargetSize: MaterialTapTargetSize.shrinkWrap,
    padding: WidgetStatePropertyAll(EdgeInsets.all(6)),
    minimumSize: WidgetStatePropertyAll(Size(32, 32)),
  );

  @override
  Widget build(BuildContext context) {
    final w = MediaQuery.sizeOf(context).width;
    final useMenu = w < 400;

    final titleText =
        '${localizeCourseName(lecture.courseName, l10n)} • ${lecture.type}${showMeetingNumber ? ' • #${effectiveMeetingNumber(lecture, allLectures)}' : ''}';
    final metaText =
        '${formatTimeRange(lecture.start, lecture.end, l10n)} • ${lecture.room} • ${statusLabelL10n(lecture.status, l10n)}';

    return Card(
      margin: const EdgeInsets.symmetric(horizontal: 0, vertical: 4),
      child: Padding(
        padding: const EdgeInsets.fromLTRB(12, 10, 12, 8),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            Row(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Container(
                  width: 5,
                  height: 40,
                  decoration: BoxDecoration(
                    color: lecture.color,
                    borderRadius: BorderRadius.circular(4),
                  ),
                ),
                const SizedBox(width: 10),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Row(
                        children: [
                          Icon(
                            meetingTypeIcon(lecture.type, l10n),
                            size: 18,
                          ),
                          const SizedBox(width: 6),
                          Expanded(
                            child: Text(
                              titleText,
                              style: Theme.of(context)
                                  .textTheme
                                  .titleSmall
                                  ?.copyWith(fontWeight: FontWeight.w600),
                              maxLines: 2,
                              overflow: TextOverflow.ellipsis,
                            ),
                          ),
                        ],
                      ),
                      const SizedBox(height: 4),
                      Text(
                        metaText,
                        style: Theme.of(context).textTheme.bodySmall?.copyWith(
                              color: Theme.of(context).colorScheme.onSurfaceVariant,
                            ),
                        maxLines: 2,
                        overflow: TextOverflow.ellipsis,
                      ),
                    ],
                  ),
                ),
              ],
            ),
            const SizedBox(height: 8),
            if (useMenu)
              Align(
                alignment: AlignmentDirectional.centerStart,
                child: _MarkMenuButton(
                  l10n: l10n,
                  lecture: lecture,
                  onStatus: onStatus,
                ),
              )
            else
              SingleChildScrollView(
                scrollDirection: Axis.horizontal,
                child: Row(
                  children: [
                    IconButton(
                      style: _compactBtnStyle,
                      tooltip: l10n.markAttended,
                      onPressed: () =>
                          onStatus(lecture, LectureStatus.attended),
                      icon: const Icon(AppIcons.attended, color: Colors.green),
                    ),
                    IconButton(
                      style: _compactBtnStyle,
                      tooltip: l10n.markMissed,
                      onPressed: () => onStatus(lecture, LectureStatus.missed),
                      icon: const Icon(AppIcons.missed, color: Colors.red),
                    ),
                    IconButton(
                      style: _compactBtnStyle,
                      tooltip: l10n.markSkipped,
                      onPressed: () =>
                          onStatus(lecture, LectureStatus.skipped),
                      icon: const Icon(AppIcons.skipped),
                    ),
                    IconButton(
                      style: _compactBtnStyle,
                      tooltip: l10n.markWatchedRecording,
                      onPressed: () => onStatus(
                        lecture,
                        LectureStatus.watchedRecording,
                      ),
                      icon: const Icon(
                        AppIcons.watchedRecording,
                        color: Colors.blue,
                      ),
                    ),
                    IconButton(
                      style: _compactBtnStyle,
                      tooltip: l10n.markCanceled,
                      onPressed: () =>
                          onStatus(lecture, LectureStatus.canceled),
                      icon: const Icon(Icons.block, color: Colors.orange),
                    ),
                  ],
                ),
              ),
          ],
        ),
      ),
    );
  }
}

class _MarkMenuButton extends StatelessWidget {
  const _MarkMenuButton({
    required this.l10n,
    required this.lecture,
    required this.onStatus,
  });

  final AppLocalizations l10n;
  final Lecture lecture;
  final void Function(Lecture, LectureStatus) onStatus;

  @override
  Widget build(BuildContext context) {
    return MenuAnchor(
      builder: (context, controller, child) {
        return FilledButton.tonalIcon(
          onPressed: () {
            if (controller.isOpen) {
              controller.close();
            } else {
              controller.open();
            }
          },
          icon: const Icon(Icons.more_horiz),
          label: Text(l10n.statusLabel),
        );
      },
      menuChildren: [
        MenuItemButton(
          leadingIcon: const Icon(AppIcons.attended, color: Colors.green),
          child: Text(l10n.markAttended),
          onPressed: () => onStatus(lecture, LectureStatus.attended),
        ),
        MenuItemButton(
          leadingIcon: const Icon(AppIcons.missed, color: Colors.red),
          child: Text(l10n.markMissed),
          onPressed: () => onStatus(lecture, LectureStatus.missed),
        ),
        MenuItemButton(
          leadingIcon: const Icon(AppIcons.skipped),
          child: Text(l10n.markSkipped),
          onPressed: () => onStatus(lecture, LectureStatus.skipped),
        ),
        MenuItemButton(
          leadingIcon:
              const Icon(AppIcons.watchedRecording, color: Colors.blue),
          child: Text(l10n.markWatchedRecording),
          onPressed: () =>
              onStatus(lecture, LectureStatus.watchedRecording),
        ),
        MenuItemButton(
          leadingIcon: const Icon(Icons.block, color: Colors.orange),
          child: Text(l10n.markCanceled),
          onPressed: () => onStatus(lecture, LectureStatus.canceled),
        ),
      ],
    );
  }
}
